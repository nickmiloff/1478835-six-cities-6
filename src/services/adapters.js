export const dataToAuthInfo = (data) => {
  const authInfo = {
    ...data,
    avatarUrl: data[`avatar_url`],
    isPro: data[`is_pro`]
  };

  delete authInfo[`avatar_url`];
  delete authInfo[`is_pro`];

  return authInfo;
};

export const dataToHotelCard = (data) => {
  const card = {
    ...data,
    host: {
      ...data.host,
      avatarUrl: data.host[`avatar_url`],
      isPro: data.host[`is_pro`]
    },
    isFavorite: data[`is_favorite`],
    isPremium: data[`is_premium`],
    previewImage: data[`preview_image`],
    maxAdults: data[`max_adults`]
  };

  delete card.host[`avatar_url`];
  delete card.host[`is_pro`];
  delete card[`is_favorite`];
  delete card[`is_premium`];
  delete card[`preview_image`];
  delete card[`max_adults`];

  return card;
};

export const dataToComment = (data) => {
  const comment = {
    ...data,
    user: {
      ...data.user,
      avatarUrl: data.user[`avatar_url`],
      isPro: data.user[`is_pro`]
    }
  };

  delete comment.user[`avatar_url`];
  delete comment.user[`is_pro`];

  return comment;
};
