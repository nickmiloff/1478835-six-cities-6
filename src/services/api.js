import * as axios from 'axios';

const BASE_URL = `https://6.react.pages.academy/six-cities/`;
const REQUEST_TIMEOUT = 5000;
const UNAUTHORIZED_CODE = 401;

export const createAPI = (onUnauthorized) => {
  const api = axios.create({
    baseURL: BASE_URL,
    headers: {
      "Content-Type": `application/json;charset=utf-8`
    },
    timeout: REQUEST_TIMEOUT,
    withCredentials: true
  });

  const onSuccess = (response) => response;

  const onFail = (err) => {
    const {response} = err;

    if (response.status === UNAUTHORIZED_CODE) {
      onUnauthorized();
      throw err;
    }

    throw err;
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
