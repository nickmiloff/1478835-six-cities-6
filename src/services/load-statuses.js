export const Statuses = {
  PENDING: `status/pending`,
  LOAD: `status/load`,
  LOADED: `status/loaded`,
  ERROR: `status/error`
};
