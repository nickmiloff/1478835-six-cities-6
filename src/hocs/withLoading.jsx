import React from 'react';
import PropTypes from 'prop-types';
import Spinner from '../components/spiner/spiner';
import * as statuses from '../services/load-statuses';

const withLoading = (Component) => {
  const WithLoading = (props) => {
    return (
      <>
        {props.isLoaded === statuses.LOAD && <Spinner />}
        <Component {...props} />
      </>
    );
  };

  WithLoading.propTypes = {
    isLoaded: PropTypes.oneOf([...Object.values(statuses)]).isRequired
  };

  return WithLoading;
};

export default withLoading;
