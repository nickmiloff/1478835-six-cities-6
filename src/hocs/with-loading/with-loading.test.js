import React from 'react';
import {render} from '@testing-library/react';
import {Statuses} from '../../services/load-statuses';

import withLoading from './with-loading';

describe(`withLoading HOC test`, () => {
  const MockComponent = () => (<span />);
  const MockComponentWithHOC = withLoading(MockComponent);

  it(`'withLoading HOC' should render correctly with 'load' status`, () => {
    const {container} = render(<MockComponentWithHOC isLoaded={Statuses.LOAD} />);

    expect(container).toMatchSnapshot();
  });

  it(`'withLoading HOC' should render correctly with 'loaded' status`, () => {
    const {container} = render(<MockComponentWithHOC isLoaded={Statuses.LOADED} />);

    expect(container).toMatchSnapshot();
  });
});
