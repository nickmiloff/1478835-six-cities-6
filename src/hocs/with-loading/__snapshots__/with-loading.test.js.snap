// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoading HOC test 'withLoading HOC' should render correctly with 'load' status 1`] = `
<div>
  <section
    class="spiner"
  >
    <div
      class="spiner__wrapper"
    >
      <div
        class="spiner__spiner"
      >
        <div />
      </div>
    </div>
  </section>
  <span />
</div>
`;

exports[`withLoading HOC test 'withLoading HOC' should render correctly with 'loaded' status 1`] = `
<div>
  <span />
</div>
`;
